### PR TITLE
Refactors to Labels, Buttons, Calculators, and ElloAPI

### PR DESCRIPTION
**Pull request by colinta**
I got a bit overzealous, but I think these are all reasonable, if they weren't all bunched up into one big PR.

The commits are tediously broken up, so going through the commit history might be easier than looking at the diff all at once.

- (big) `ElloLabel` and friends is gone, replaced with `StyledLabel` (consistency changes to `StyledButton`, too)
- (small) Calculators moved into `Calculators/` folder, and some style changes to `StreamNotificationCellSizeCalculator`
- (fix) fixes UserSpec (`formattedShortBio`)
- Only fetch "current" dSyms, so the fastlane `refresh_dsyms` command is more useful, and run automatically
- Minor fix to LoginScreen when displaying errors
- ElloAPI refactors (code consistency, 'createdAt' header is optional in new-content endpoints)
- Fix to scrolling in Notifications - see the 'contentInset' hack
- When clearing out placeholders, instead of deleting them all, an empty placeholder is inserted (future-bug-fix, needed in Announcements branch)
- Check out how `NSAttributedString` can accept `StyledLabel.Style` or `StyledButton.Style` - makes the calculators easy to write.
- Added the `afterN` - this actually comes from my game code - allows for arbitrary/unknown/non-magical 'after' behavior.
- `uniqId` is now `tableId` (used in YAP) and `uniqueId` (used to compare *any* two `JSONAbles`)
- Changed `LoveChangedNotification` to generic `JSONAbleChangedNotification`, so that the same notification can be used in Announcements
- Lots of misc. code style changes

### CODE REVIEW

File: Sources/Views/ElloButton.swift [hunk 1]
```diff
@@ -1,206 +0,0 @@
-////
```
Comment (steam) [inline Sources/Views/ElloButton.swift]: Yeah RED!

File: Sources/Controllers/Profile/ProfileViewController.swift [hunk 2]
```diff
@@ -71,9 +71,9 @@ public final class ProfileViewController: StreamableViewController {
 
     private func sharedInit() {
         streamViewController.streamKind = initialStreamKind
-        streamViewController.initialLoadClosure = { [unowned self] in self.loadProfile() }
-        streamViewController.reloadClosure = { [unowned self] in self.reloadEntireProfile() }
-        streamViewController.toggleClosure = { [unowned self] isGridView in self.toggleGrid(isGridView) }
+        streamViewController.initialLoadClosure = { [weak self] in self?.loadProfile() }
```
Comment (steam) [inline Sources/Controllers/Profile/ProfileViewController.swift]: 🐛 🔨 🤘

File: Sources/Views/ElloCollectionView.swift [hunk 3]
```diff
@@ -5,5 +5,32 @@
...
+// This `contentInset` nonsense is to prevent UIKit from setting the content
+// inset when we don't want it to.  Setting `automaticallyAdjustsContentInsets`
+// to `false` *should* prevent this from happening, but it doesn't.  So instead,
+// the `StreamViewController` sets `elloContentInset`.  If `contentInset` is
+// set outside the scope of `uikitOverride` being set, it will be *restored* to
+// whatever `elloContentInset` was set to.
+//
+// UG
+//
 public class ElloCollectionView: UICollectionView {
```
Comment (steam) [inline Sources/Views/ElloCollectionView.swift]: This reminds me a bit of the issue I found yesterday where reloading a cell in a table view that automatically sizes it's cells causes the scrollview to scroll to the top. :(

File: Sources/Controllers/Stream/Calculators/CategoryHeaderCellSizeCalculator.swift [hunk 4]
```diff
@@ -10,7 +10,7 @@ public class CategoryHeaderCellSizeCalculator {
 
     private typealias CellJob = (cellItems: [StreamCellItem], width: CGFloat, completion: ElloEmptyCompletion)
     private var cellJobs: [CellJob] = []
-    private var screenWidth: CGFloat = 0.0
+    private var cellWidth: CGFloat = 0.0
```
Comment (steam) [inline Sources/Controllers/Stream/Calculators/CategoryHeaderCellSizeCalculator.swift]: 👍

File: Sources/Controllers/Stream/StreamViewController.swift [hunk 5]
```diff
@@ -326,6 +326,11 @@ public final class StreamViewController: BaseElloViewController {
         completion: ElloEmptyCompletion = {}
         )
     {
+        guard streamCellItems.count > 0 else {
+            replacePlaceholder(placeholderType, with: [StreamCellItem(type: .Placeholder, placeholderType: placeholderType)], completion: completion)
```
Comment (steam) [inline Sources/Controllers/Stream/StreamViewController.swift]: You mentioned this yesterday. Nice approach. Remind me again when this was happening.

File: Sources/Utilities/FreeMethods.swift [hunk 6]
```diff
@@ -108,6 +108,37 @@ private func times_(times: Int, @noescape block: TakesIndexBlock) {
...
+// func networkCalls(completion: BasicBlock) {
+//     let (afterAll, done) = afterN() { completion() }
+//     backgroundProcess1(completion: afterAll())
+//     backgroundProcess2(completion: afterAll())
+//     done()  // this doesn't execute the callback, just says "i'm done registering callbacks"
+// }
+//
+// without this 'done' trick, there is a bug where if the first process is synchronous, the 'count'
+// is incremented (by calling 'afterAll') and then immediately decremented.
+public func afterN(_ block: BasicBlock) -> (() -> BasicBlock, BasicBlock) {
```
Comment (steam) [inline Sources/Utilities/FreeMethods.swift]: 👌

File: Sources/Controllers/Stream/CellDequeing/StreamToggleCellPresenter.swift [hunk 7]
```diff
@@ -13,11 +13,18 @@ public struct StreamToggleCellPresenter {
         indexPath: NSIndexPath,
         currentUser: User?)
     {
-        if let cell = cell as? StreamToggleCell {
-            if let post = streamCellItem.jsonable as? Post {
-                let message = streamCellItem.state == .Collapsed ? cell.closedMessage : cell.openedMessage
-                cell.label.setLabelText(post.contentWarning + " " + message)
-            }
+        guard let
```
Comment (colinta) [inline Sources/Controllers/Stream/CellDequeing/StreamToggleCellPresenter.swift]: Do we not have NSFW filtering any more?  The cell is used in `StreamCellItemParser.swift`:

```swift
private func postToggleItems(post: Post) -> [StreamCellItem] {
    if post.collapsed {
        return [StreamCellItem(jsonable: post, type: .Toggle)]
    }
    else {
        return []
    }
}
```

File: Sources/Controllers/Stream/StreamViewController.swift (hunk 5 above)
Comment (colinta) [inline Sources/Controllers/Stream/StreamViewController.swift]: It's helpful in announcements, where it's easy to have "zero announcement cells", but if a "new announcements avail" notification comes in (triggering a "load announcements network call), the `replacePlaceholders` still needs to have >0 cells there to replace.

**Comment by colinta** [comment]
Rebased onto master, and update the `OmnibarErrorCell` to use `InterfaceString`.

**Comment by colinta** [comment]
Cool, actually got a green check from travis! (not from both pr & push, oddly)

**Pull request merged**